autosubmit/git/autosubmit_git.py

```diff
@@ -45,6 +42,45 @@
     )''', re.VERBOSE)
 """Regular expression to match Git URL."""
 
+
+# M: Modified, A: Added, D: Deleted, ?: Untracked
+_GIT_STATUS_DIRTY_FLAG = ["M", "A", "D", "?"]
+"""List of git statuses. The output of ``git status --porcelain`` contain
+flags to indicate the status of the object."""
+
+
+def check_unpushed_changes(expid: str, as_conf: AutosubmitConfig) -> None:
+    """Check if the Git repository is dirty for an operational experiment.
+
+    Raises an AutosubmitCritical error if the experiment is operational,
+    the platform is Git, and there are unpushed changes in the local Git
+    repository, or in any of its Git submodules.
+
+    :param expid: The experiment ID.
+    :param as_conf: Autosubmit configuration object.
+    """
+    project_type = as_conf.get_project_type()
+    if expid[0] == 'o' and project_type == 'git':
+        git_cmds = [
+            ["git"],
+            ["git", "submodule"]
+        ]
+        arguments = [
+            ["status", "--porcelain"],
+            ["foreach", "'git status --porcelain'"]
+        ]
+
+        proj_dir = as_conf.get_project_dir()
+
+        for git_cmd, args in zip(git_cmds, arguments):
+            output = subprocess.check_output([*git_cmd, *args], cwd=proj_dir)
```

Comment:
Deconstructing the arrays here (this `*`), from what I recall, you can pass a string or an array, but not mixing them. So now everything should be just an array of strings. This was one of the bugs we had.